pilot/pkg/networking/core/v1alpha3/configgen.go

```diff
@@ -37,7 +41,9 @@ func NewConfigGenerator(plugins []plugin.Plugin) *ConfigGeneratorImpl {
 }
 
 func (configgen *ConfigGeneratorImpl) BuildSharedPushState(env *model.Environment, push *model.PushContext) error {
-	configgen.PrecomputedOutboundClusters = configgen.buildOutboundClusters(env, nil, push)
+	configgen.mutex.Lock()
+	defer configgen.mutex.Unlock()
+	configgen.publicClusters, configgen.privateClustersByNamespace = configgen.buildOutboundClusters(env, nil, push)
```

Comment:
This is not the right way to do it. What we need is a list of all the proxy config namespaces, such that during the BuildSharedPushState, we build this state for each config namespace (create a dummy node with config namespace set). I suggest removing this logic for now. I am working on adding it in another PR..

Reply:
You mean record clusters map[ns][]*xdsapi.Cluster?